Incompatible with latest casacore

As reported by @revoltek:
```
#################
/home/fdg/opt/lofar_190705/DPPP/src/Common/BaselineSelect.cc: In static member function ‘static casacore::Matrix<bool> DP3::BaselineSelect::convert(casacore::Table&, casacore::TableExprNode&, casacore::TableExprNode&, const string&, std::ostream&)’:
/home/fdg/opt/lofar_190705/DPPP/src/Common/BaselineSelect.cc:121:59: error: cannot convert ‘casacore::CountedPtr<casacore::MSSelectionErrorHandler>’ to ‘casacore::MSSelectionErrorHandler*’ in initialization
    MSSelectionErrorHandler* curHandler = MSAntennaParse::thisMSAErrorHandler;
                                                          ^
make[2]: *** [CMakeFiles/Common.dir/Common/BaselineSelect.cc.o] Error 1
make[1]: *** [CMakeFiles/Common.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
/home/fdg/opt/lofar_190705/DPPP/src/DPPP/MSWriter.cc: In member function ‘void DP3::DPPP::MSWriter::createMS(const string&, const DP3::DPPP::DPInfo&, unsigned int, unsigned int)’:
/home/fdg/opt/lofar_190705/DPPP/src/DPPP/MSWriter.cc:286:35: error: ‘SimpleOrderedMap’ was not declared in this scope
      newdesc.adjustHypercolumns (SimpleOrderedMap<String,String>(String()));
                                  ^
/home/fdg/opt/lofar_190705/DPPP/src/DPPP/MSWriter.cc:286:58: error: expected primary-expression before ‘,’ token
      newdesc.adjustHypercolumns (SimpleOrderedMap<String,String>(String()));
                                                         ^
/home/fdg/opt/lofar_190705/DPPP/src/DPPP/MSWriter.cc:286:65: error: expected primary-expression before ‘>’ token
      newdesc.adjustHypercolumns (SimpleOrderedMap<String,String>(String()));
                                                                ^
make[2]: *** [CMakeFiles/DPPP_OBJ.dir/DPPP/MSWriter.cc.o] Error 1
make[1]: *** [CMakeFiles/DPPP_OBJ.dir/all] Error 2
make: *** [all] Error 2
###################
```

--- a/Common/BaselineSelect.cc
+++ b/Common/BaselineSelect.cc
@@ -118,7 +118,7 @@ namespace DP3 {
     Vector<Int> selectedAnts1;
     Vector<Int> selectedAnts2;
     Matrix<Int> selectedBaselines;
-    CountedPtr<MSSelectionErrorHandler> curHandler = MSAntennaParse::thisMSAErrorHandler;
+    auto curHandler = MSAntennaParse::thisMSAErrorHandler;
     BaselineSelectErrorHandler errorHandler (os);
     MSAntennaParse::thisMSAErrorHandler = &errorHandler;
     try {

--- a/Common/BaselineSelect.cc
+++ b/Common/BaselineSelect.cc
@@ -118,7 +118,7 @@ namespace DP3 {
     Vector<Int> selectedAnts1;
     Vector<Int> selectedAnts2;
     Matrix<Int> selectedBaselines;
-    MSSelectionErrorHandler* curHandler = MSAntennaParse::thisMSAErrorHandler;
+    CountedPtr<MSSelectionErrorHandler> curHandler = MSAntennaParse::thisMSAErrorHandler;
     BaselineSelectErrorHandler errorHandler (os);
     MSAntennaParse::thisMSAErrorHandler = &errorHandler;
     try {

--- a/DPPP/MSWriter.cc
+++ b/DPPP/MSWriter.cc
@@ -43,6 +43,7 @@
 #include <casacore/casa/Arrays/ArrayLogical.h>
 #include <casacore/casa/Containers/Record.h>
 #include <casacore/casa/OS/Path.h>
+#include <casacore/casa/version.h>
 
 #include <iostream>
 #include <limits>
@@ -283,7 +284,13 @@ namespace DP3 {
         }
       }
       // Remove possible hypercolumn definitions.
+// Test for casacore version 3.1.1 or smaller
+#pragma message "BOOST_VERSION=" BOOST_PP_STRINGIZE(CASACORE_VERSION_NUM)
+#if CASACORE_MAJOR_VERSION<3 || (CASACORE_MAJOR_VERSION==3 && (CASACORE_MINOR_VERSION==0 || (CASACORE_MINOR_VERSION==1 && CASACORE_PATCH_VERSION < 2)))
       newdesc.adjustHypercolumns (SimpleOrderedMap<String,String>(String()));
+#else
+      newdesc.adjustHypercolumns (std::map<String,String>());
+#endif
       // Set data manager info.
       Record dminfo = temptable.dataManagerInfo();
       // Determine the DATA tile shape. Use all corrs and the given #channels.

